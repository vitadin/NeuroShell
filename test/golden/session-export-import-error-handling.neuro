%% Test error handling for export/import commands
\set[_echo_command="true"]

%% Test export with missing file parameter
\try \session-json-export nonexistent_session

%% Test export with non-existent session
\try \session-json-export[file=/tmp/test.json] completely_nonexistent_session

%% Test import with missing file parameter  
\try \session-json-import

%% Test import with non-existent file
\try \session-json-import[file=/tmp/completely_nonexistent_file.json]

%% Test import with invalid JSON file
\try \session-json-import[file=test/golden/session-export-import-error-handling/invalid.json]

%% Test general import with unsupported file extension (should fail gracefully)
\try \session-import[file=/tmp/test.txt]

%% Verify _output variables contain error information for the failed attempts
\get[_output]