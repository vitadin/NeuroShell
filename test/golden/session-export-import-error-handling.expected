Setting _echo_command = true
%%> "\\try \\session-json-export nonexistent_session"
%%> "\\session-json-export nonexistent_session"
%%> "\\try \\session-json-export[file=/tmp/test.json] completely_nonexistent_session"
%%> "\\session-json-export[file=/tmp/test.json] completely_nonexistent_session"
%%> "\\try \\session-json-import"
%%> "\\session-json-import"
%%> "\\try \\session-json-import[file=/tmp/completely_nonexistent_file.json]"
%%> "\\session-json-import[file=/tmp/completely_nonexistent_file.json]"
%%> "\\try \\session-json-import[file=test/golden/session-export-import-error-handling/invalid.json]"
%%> "\\session-json-import[file=test/golden/session-export-import-error-handling/invalid.json]"
%%> "\\try \\session-import[file=/tmp/test.txt]"
%%> "\\session-import[file=/tmp/test.txt]"
%%> "\\get[_output]"
_output = 